gcloud-java-compute/src/main/java/com/google/gcloud/compute/spi/DefaultComputeRpc.java

```diff
@@ -869,6 +880,256 @@ public Operation deleteNetwork(String network, Map<Option, ?> options) {
     }
   }
 
+  @Override
+  public Operation createInstance(String zone, Instance instance, Map<Option, ?> options) {
+    try {
+      return compute.instances()
+          .insert(this.options.projectId(), zone, instance)
+          .setFields(FIELDS.getString(options))
+          .execute();
+    } catch (IOException ex) {
+      throw translate(ex);
+    }
+  }
+
+  @Override
+  public Instance getInstance(String zone, String instance, Map<Option, ?> options) {
+    try {
+      return compute.instances()
+          .get(this.options.projectId(), zone, instance)
+          .setFields(FIELDS.getString(options))
+          .execute();
+    } catch (IOException ex) {
+      return nullForNotFound(ex);
+    }
+  }
+
+  @Override
+  public Tuple<String, Iterable<Instance>> listInstances(String zone, Map<Option, ?> options) {
+    try {
+      InstanceList instanceList = compute.instances()
+          .list(this.options.projectId(), zone)
+          .setFilter(FILTER.getString(options))
+          .setMaxResults(MAX_RESULTS.getLong(options))
+          .setPageToken(PAGE_TOKEN.getString(options))
+          .setFields(FIELDS.getString(options))
+          .execute();
+      Iterable<Instance> instances = instanceList.getItems();
+      return Tuple.of(instanceList.getNextPageToken(), instances);
+    } catch (IOException ex) {
+      throw translate(ex);
+    }
+  }
+
+  @Override
+  public Tuple<String, Iterable<Instance>> listInstances(Map<Option, ?> options) {
+    try {
+      InstanceAggregatedList aggregatedList = compute.instances()
+          .aggregatedList(this.options.projectId())
+          .setFilter(FILTER.getString(options))
+          .setMaxResults(MAX_RESULTS.getLong(options))
+          .setPageToken(PAGE_TOKEN.getString(options))
+          // todo(mziccard): uncomment or remove once #711 is closed
+          // .setFields(FIELDS.getString(options))
+          .execute();
+      ImmutableList.Builder<Instance> builder = ImmutableList.builder();
+      Map<String, InstancesScopedList> scopedList = aggregatedList.getItems();
+      if (scopedList != null) {
+        for (InstancesScopedList instancesScopedList : scopedList.values()) {
+          if (instancesScopedList.getInstances() != null) {
+            builder.addAll(instancesScopedList.getInstances());
+          }
+        }
+      }
+      return Tuple.<String, Iterable<Instance>>of(aggregatedList.getNextPageToken(),
+          builder.build());
+    } catch (IOException ex) {
+      throw translate(ex);
+    }
+  }
+
+  @Override
+  public Operation deleteInstance(String zone, String instance, Map<Option, ?> options) {
+    try {
+      return compute.instances()
+          .delete(this.options.projectId(), zone, instance)
+          .setFields(FIELDS.getString(options))
```

Comment:
Delete returns an `Operation` resource and we can select which fields of such operation resource we want to return, for instance you might just be interested in the operation's id.